commit 121a46a026afac197652cb8a4cdf2d3879d7f587
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Mon Oct 10 09:34:53 2022 -0300

    btf_encoder: Store the CU being processed to avoid changing many functions
    
    Having it as encoder->cu will make it available to nested function
    without requiring changing all the functions leading to them.
    
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

---
 btf_encoder.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

Index: b/btf_encoder.c
===================================================================
--- a/btf_encoder.c
+++ b/btf_encoder.c
@@ -44,9 +44,13 @@ struct var_info {
 	uint32_t    sz;
 };
 
+/*
+ * cu: cu being processed.
+ */
 struct btf_encoder {
 	struct list_head  node;
 	struct btf        *btf;
+	struct cu         *cu;
 	struct gobuffer   percpu_secinfo;
 	const char	  *filename;
 	struct elf_symtab *symtab;
@@ -1232,8 +1236,9 @@ static bool ftype__has_arg_names(const s
 	return true;
 }
 
-static int btf_encoder__encode_cu_variables(struct btf_encoder *encoder, struct cu *cu, uint32_t type_id_off)
+static int btf_encoder__encode_cu_variables(struct btf_encoder *encoder, uint32_t type_id_off)
 {
+	struct cu *cu = encoder->cu;
 	uint32_t core_id;
 	struct tag *pos;
 	int err = -1;
@@ -1465,6 +1470,7 @@ int btf_encoder__encode_cu(struct btf_en
 	struct tag *pos;
 	int err = 0;
 
+	encoder->cu = cu;
 
 	if (!encoder->has_index_type) {
 		/* cu__find_base_type_by_name() takes "type_id_t *id" */
@@ -1580,8 +1586,9 @@ int btf_encoder__encode_cu(struct btf_en
 	}
 
 	if (!encoder->skip_encoding_vars)
-		err = btf_encoder__encode_cu_variables(encoder, cu, type_id_off);
+		err = btf_encoder__encode_cu_variables(encoder, type_id_off);
 out:
+	encoder->cu = NULL;
 	return err;
 }
 
